react/src/component/simulation.tsx

```diff
@@ -0,0 +1,263 @@
+import {
+    Button,
+    Col,
+    Container, Dropdown, Form, Modal, Row
+} from "react-bootstrap";
+import React, {
+    useState,
+    useEffect,
+    useContext
+} from "react";
+import {
+    modelSelectors,
+    modelActions,
+    ModelState,
+    ModelStates
+} from "../store/models";
+import { FormStateInitializer } from "./reusable/form";
+import { useNavigate, useResolvedPath } from "react-router-dom";
+import { CRelativeRouterHelper, RouteHelper } from "../utility/route";
+import { ReportEventManager } from "../data/report";
+import { CReportEventManager } from "../data/report";
+import { CModelsWrapper, ModelsWrapper } from "../data/wrapper";
+import { AppWrapper, CAppName, CSchema, CSimulationInfoPromise } from "../data/appwrapper";
+import { LAYOUTS } from "../layout/layouts";
+import { ModelsAccessor } from "../data/accessor";
+import { Dependency } from "../data/dependency";
+import { NavbarRightContainerId, NavToggleDropdown } from "./reusable/navbar";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import * as Icon from "@fortawesome/free-solid-svg-icons";
+import { useSetup } from "../hook/setup";
+import { Portal } from "./reusable/portal";
+
+export type SimulationInfoRaw = {
+    models: ModelStates,
+    simulationType: string,
+    version: string
+}
+
+export type SimulationInfo = SimulationInfoRaw & {
+    simulationId: string
+}
+
+function SimulationInfoInitializer(props: { simulationId: string } & {[key: string]: any}) {
+    let { simulationId } = props;
+
+    let [simulationInfoPromise, updateSimulationInfoPromise] = useState(undefined);
+    let [hasInit, updateHasInit] = useState(false);
+    let appName = useContext(CAppName);
+
+    let modelsWrapper = new ModelsWrapper({
+        modelActions,
+        modelSelectors
+    })
+
+    useEffect(() => {
+        updateSimulationInfoPromise(new Promise((resolve, reject) => {
+            // TODO: why 0
+            fetch(`/simulation/${appName}/${simulationId}/0/source`).then(async (resp) => {
+                let simulationInfo = await resp.json();
+                let models = simulationInfo['models'] as ModelState[];
+
+                for(let [modelName, model] of Object.entries(models)) {
+                    modelsWrapper.updateModel(modelName, model);
+                }
+
+                resolve({...simulationInfo, simulationId});
+                updateHasInit(true);
```

Comment:
There has to be a better way to manage state than cascading booleans. This relates to the state machine problem for messages.